core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -458,10 +457,12 @@ class GroupCoordinator(val brokerId: Int,
   }
 
   private def addMemberAndRebalance(sessionTimeoutMs: Int,
+                                    clientId: String,
                                     protocols: List[(String, Array[Byte])],
                                     group: GroupMetadata,
                                     callback: JoinCallback) = {
-    val memberId = group.generateNextMemberId
+    // use the client-id with a random id suffix as the member-id
+    val memberId = clientId + "-" + group.generateMemberIdSuffix
```

Comment:
Yeah there is a separate ticket KAFKA-2018, I was just trying to tackle it in this patch as well for shorter review cycle since it is small change.